src/Microsoft.Sbom.Api/Manifest/Configuration/SbomConfigProvider.cs

```diff
@@ -118,15 +118,22 @@ public IDisposable StartJsonSerialization()
         return this;
     }
 
-    public IAsyncDisposable StartJsonSerializationAsync()
+    /// <summary>
+    /// Starts asynchronous JSON serialization of supported ISbomConfig objects from IConfiguration.
+    /// </summary>
+    /// <param name="manifestInfosFromConfig"></param>
+    /// <returns></returns>
+    public IAsyncDisposable StartJsonSerializationAsync(IList<ManifestInfo> manifestInfosFromConfig)
     {
-        ApplyToEachConfig(c => c.StartJsonSerialization());
-        return this;
-    }
+        foreach (var manifestInfo in manifestInfosFromConfig)
+        {
+            var supportedManifestInfo = this.TryGet(manifestInfo, out var config);
```

Comment:
It's common to leverage the guarded block for scoping, which ensures that it can't accidentally be used outside the block where it is valid:

```
if (TryGet(manifestInfo, out var config))
{
  config.StartJsonSerialization();
}
```